lib/commands/waitForExist.js

```diff
@@ -27,7 +29,13 @@ module.exports = function waitForExist(selector, ms, reverse) {
         reverse = false;
     }
 
+    var start = Date.now();
+
     return this.waitUntil(function() {
+        if (Date.now() > start + ms) {
+            // throw NoSuchElement error if no element was found before timeout
+            throw new ErrorHandler(7);
+        }
```

Comment:
waitUntil takes care of this: https://github.com/webdriverio/webdriverio/blob/master/lib/commands/waitUntil.js#L66-L72